pennylane/queuing.py

```diff
@@ -11,8 +11,115 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""
-This module contains the :class:`QueuingManager`.
+r"""
+This module contains the classes for placing objects into queues.
+
+Description
+-----------
+
+A *queuable object* is anything that can be placed into a queue. These will be :class:`~.Operator`,
+:class:`~.MeasurementProcess`, and :class:`~.QuantumTape` objects. :class:`~.Operator` and
+:class:`~.MeasurementProcess` objects achieve queuing via a ``queue`` method called upon construction.
+Note that even though :class:`~.QuantumTape` is a queuable object, it does not have a ``queue`` method.
+
+When an object is queued, it sends itself to the :class:`~.QueuingManager`. The :class:`~.QueuingManager`
+is a global singleton class that facilitates placing objects in the queue. All of :class:`~.QueuingManager`'s methods
+and properties are class methods and properties, so all instances will access the same information.
+
+The :meth:`~.QueuingManager.active_context` is the queue where any new objects are placed.
+The :class:`~.QueuingManager` is said to be *recording* if an active context exists.
+
+Active contexts are :class:`~.AnnotatedQueue` instances. They are *context managers* where recording occurs
+within a ``with`` block.
+
+Let's take a look at an example. If we query the :class:`~.QueuingManager` outside of an
+:class:`~.AnnotatedQueue`'s context, we can see that nothing is recording and no active context exists.
+
+>>> print("Are we recording? ", qml.QueuingManager.recording())
+Are we recording?  False
+>>> print("What's the active context?", qml.QueuingManager.active_context())
+What's the active context? None
```

Comment:
```suggestion
>>> print("What's the active context? ", qml.QueuingManager.active_context())
What's the active context?  None
```
just to be consistent